### DISCUSSION TITLE
XML error "Content is not allowed in prolog"

### DISCUSSION THREAD

**Discussion by apple162883** [nextgenhealthcare/connect#6168] [Q&A]
Using Mirth Connect 4.2.0 to read an XML File.
The Source Connector Inbound is set to XML. Using Batch Processing and splitting the Batch by Element Name

The inbound file contains UTF 8 BOM {ISO-8859-1} character 
{ï»¿<?xml version="1.0" encoding="UTF-8">} 
in the first line which causes it to throw
 "Content is not allowed in prolog" error.

I tried 
return message.replace(/\uFEFF/g,'') in the Preprocessor but it seems code does not reach upto that line and throws error.

Any suggestions to bypass or remove this character so I can parse the XML File 

Thank you in advance

**Comment by pacmano1** [comment]
https://forums.mirthproject.io/forum/mirth-connect/support/17964-utf-8-bom-remove-special-characters-xml-reader

**Reply by apple162883** [reply]
The above solution only works when i disable the Batch Processing.

When I enable batch processing and split batches by Element it still throws "Content is not allowed in prolog".
Actually I doubt the process even comes to Preprocessor as I am not able print any logger before the return statement in Preprocessor.

Any suggestion on how to handle this while Batch Processing Mode is ON
